http/handler.go

```diff
@@ -86,7 +86,15 @@ func NewHandler(db client.DB) (*Handler, error) {
 		r.Handle("/*", router)
 	})
 	mux.Get("/openapi.json", func(rw http.ResponseWriter, req *http.Request) {
-		responseJSON(rw, http.StatusOK, router.OpenAPI())
+		oas := router.OpenAPI()
+		addr, _ := req.Context().Value(serverAddrKey).(string)
+		isTLS, _ := req.Context().Value(serverTLSKey).(bool)
+		if isTLS {
+			oas.Servers[0].URL = "https://" + addr + "/api/" + Version
+		} else {
+			oas.Servers[0].URL = "http://" + addr + "/api/" + Version
+		}
+		responseJSON(rw, http.StatusOK, oas)
```

Comment:
thought: I'm very sorry Chris. After Keenan suggested to use the `req.Host` field I spent a bit of time thinking about this and I'm realizing that what we are doing is futile. There are a couple reasons for this but the most important one it that there are good chances that Defra will be deployed behind a proxy. This means that `isTLS` being false has no relationship to the client using a TLS connection or not. Also, there are no guarantees that `req.Host` will be populated with the correct address. So regardless of how smart we try to be about it, we will never be able to accurately predict the correct URL for all deployment scenarios.  The only thing that is predictable here is the path of the API. As such, this is my suggestion:

suggestion: Leave everything as it was (remove all new code) except for the URL of the OpenAPI server in `http/openapi.go` (see bellow).

Reply:
I understand, and this makes sense. I have now removed these changes.